lightning/src/chain/keysinterface.rs

```diff
@@ -1134,12 +1145,30 @@ impl KeysInterface for KeysManager {
 		self.inbound_payment_key.clone()
 	}
 
-	fn get_destination_script(&self) -> Script {
-		self.destination_script.clone()
+	fn get_destination_script(&self, channel_keys_id : [u8; 32]) -> Script {
+		let derivation_idx = if self.new_style_derivation {
+			byte_utils::slice_to_be64(&channel_keys_id[0..8])
+		} else {
+			3
+		};
+		match ExtendedPrivKey::new_master(Network::Testnet, &self.seed) {
+			Ok(master_key) => {
+				match master_key.ckd_priv(&self.secp_ctx, ChildNumber::from_hardened_idx(derivation_idx as u32).expect("key space exhausted")){
```

Comment:
Hmm, I actually liked the two-level derivation - it seems like kinda strange to have two different sets of derivations against the master key - one with a few fixed indexes and one with a random number. I'd strongly prefer to just create a new fixed index (I think 4 is free?) and do the new-style derivations off of that. Obvious the old style derivations will need to stay where they are (fixed at 3 directly off the master).

Reply:
so basically derive from the channel_master_key at either 3 or 4 (4 for new style), and then derive the byte slice from the channel_keys_id if it's new style?

Reply:
Yep! That sounds great to me.

Reply:
@TheBlueMatt I'm assuming that we'll have to use and derive the channel master key on the spending side as well? I tried out the changes and have some tests failing 😞

Reply:
I basically have the assertion for the shutdown pubkey failing (2 tests).